crowbar_framework/app/models/service_object.rb

```diff
@@ -1540,6 +1524,79 @@ def wait_for_chef_daemons(node_list)
 
   private
 
+  THREAD_POOL_SIZE = 20
+
+  def release_chef_locks(locks)
+    return if locks.empty?
+
+    queue = Queue.new
+    locks.each { |l| queue.push l }
+
+    workers = (1...THREAD_POOL_SIZE).map do
```

Comment:
I am sure that I am misunderstanding something. IIUC the original code crate batchs of 20 threads, but I can see that this iterate over THREAD_POOL_SIZE, without empty the queue (that can be bigger than 20). I do not see this also in the `lock_and_wait_for_chef_clients` method